packages/cli/src/services/annotationTag.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Service } from 'typedi';
+import { validateEntity } from '@/GenericHelpers';
+import type { ITagWithCountDb } from '@/Interfaces';
+import type { AnnotationTagEntity } from '@db/entities/AnnotationTagEntity';
+import { AnnotationTagRepository } from '@db/repositories/annotationTag.repository';
+import { ExternalHooks } from '@/ExternalHooks';
+
+type GetAllResult<T> = T extends { withUsageCount: true }
+	? ITagWithCountDb[]
+	: AnnotationTagEntity[];
+
+@Service()
+export class AnnotationTagService {
+	constructor(
+		private externalHooks: ExternalHooks,
+		private tagRepository: AnnotationTagRepository,
+	) {}
+
+	toEntity(attrs: { name: string; id?: string }) {
+		attrs.name = attrs.name.trim();
+
+		return this.tagRepository.create(attrs);
+	}
+
+	async save(tag: AnnotationTagEntity, actionKind: 'create' | 'update') {
+		await validateEntity(tag);
+
+		const action = actionKind[0].toUpperCase() + actionKind.slice(1);
+
+		await this.externalHooks.run(`annotationTag.before${action}`, [tag]);
```

Comment:
I wonder if we want to add external hooks even tho we don't know anyone using them.. 🤔 Asked about this in [Slack](https://n8nio.slack.com/archives/C069HS026UF/p1724321574896769)

Reply:
I removed the external hooks as there is no current requirement for them.